crates/nu-command/src/filters/each.rs

```diff
@@ -144,10 +155,14 @@ with 'transpose' first."#
         let redirect_stdout = call.redirect_stdout;
         let redirect_stderr = call.redirect_stderr;
 
+        // Q: Does this have to use a match expression, as compared to
+        // other loop commands like `any` or `all`, which also operate on lists or single values?
```

Comment:
Yeah, I think `match` makes sense here for now. IMO `any`/`all`/etc. are too liberal in what they accept for input and that's mostly accidental: https://github.com/nushell/nushell/issues/6941#issuecomment-1312853690

In the future, maybe we should write an alternative to `into_iter()`+`into_interruptible_iter()` that only returns an iterator for types that are actually iterable. `try_into_iter()` or something like that.

Reply:
The thing is, though, at the bottom of those `match` blocks is a match for `PipelineData::Value(x, ..)`, meaning that it accepts non-iterators anyway.